src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -246,9 +246,16 @@ public final void apply(Project project) {
                 configureAllProjectsUsingConstraints(project, rootLockfile, lockedConfigurations);
             });
         } else {
-            // projectsEvaluated is necessary to ensure all projects' dependencies have been configured, because we
+            // afterEvaluate is necessary to ensure all projects' dependencies have been configured, because we
             // need to copy them eagerly before we add the constraints from the lock file.
-            project.getGradle().projectsEvaluated(g -> {
+
+            // Note: do not use Gradle.projectsEvaluated here, as IntelliJ's [gradle import action][1] happens BEFORE
+            // that, and will thus resolve the locked configurations before we get a change to add constraints to them.
+            // [1]: https://github.com/JetBrains/intellij-community/commit/f394c51cff59c69bbaf63a8bf67cefbad9e357aa#diff-04b9936e4249a0f5727414555b76c4b9R123
+            project.afterEvaluate(g -> {
+                // Ensure all other projects have been resolved before.
+                project.getSubprojects().forEach(subproject -> project.evaluationDependsOn(subproject.getPath()));
```

Comment:
Why does this need to live inside the afterEvaluate block? Shouldn't you mention this is for conjure?

Reply:
yep indeed, it's for conjure

Reply:
sooo javadoc? or tests?

Reply:
evaluationDependsOn is eager so it's safer to delay it until afterEvaluate in case child projects depend on _some_ stuff from the root project having been configured (e.g. version)